python/hail/expr.py

```diff
@@ -26,6 +27,7 @@ class Type(object):
 
     def __init__(self, jtype):
         self._jtype = jtype
+        self._history = None
```

Comment:
That wasn't my intention -- I just wanted to see "TDouble()" instead of "Double". Your way is better. I hadn't thought of tracking the source of schemas. I'll remove the history on Type for now because I agree it is probably unnecessary.